packages/core/upload/shared/contracts/folders.ts

```diff
@@ -0,0 +1,169 @@
+import type { Data, Modules } from '@strapi/types';
+import { errors } from '@strapi/utils';
+
+export type SortOrder = 'ASC' | 'DESC';
+
+export type SortKey = 'createdAt' | 'name' | 'updatedAt';
+
+export interface FolderStructure {
+  id?: Data.ID;
+  name?: string;
+  children?: [] | FolderStructure[];
+}
+
+/**
+ * GET /upload/folder-structure
+ *
+ * Return the structure of a folder.
+ */
+export declare namespace FolderStructureNamespace {
+  export interface Request {
+    query?: {};
+  }
+
+  export interface Response {
+    data: {
+      data: FolderStructure[];
+    };
+    error?: errors.ApplicationError | errors.NotFoundError;
+  }
+}
+
+/**
+ * GET /upload/folders - Get folders
+ */
+export interface Folder {
+  name: string;
+  id: Data.ID;
+  documentId: Modules.Documents.ID;
+  pathId: Data.ID;
+  path: `/${Data.ID}`;
+  createdAt: string;
+  createdBy?: string;
+  updatedAt: string;
+  updatedBy?: string;
+  publishedAt: string | null;
+  locale: string | null;
+  children?: {
+    count: number;
+  };
+  files?: {
+    count: number;
+  };
+}
+
+export interface FolderEnriched extends Folder {
```

Comment:
Could you find a name that indicates in which way it's enriched? Or where it's intended to be used? Enriched could mean lots of things